synapse/handlers/federation.py

```diff
@@ -1742,3 +1737,21 @@ async def _sync_partial_state_room(
                             room_id,
                             destination,
                         )
+
+def _prioritise_destinations_for_partial_state_resync(
+    initial_destination: Optional[str], other_destinations: Collection[str]
+) -> Sequence[str]:
+    """Work out the order in which we should ask servers to resync events.
+
+    If an `initial_destination` is given, it takes top priority. Otherwise
+    all servers are treated equally.
+    """
+    if initial_destination is not None:
```

Comment:
suggest inverting this condition for simplicity:

```suggestion
    if initial_destination is None:
        return other_destinations
```

Reply:
1c9988ee36